src/platform-implementation-js/dom-driver/gmail/views/gmail-nav-item-view.ts

```diff
@@ -611,7 +611,7 @@ export default class GmailNavItemView {
         buttonOptions,
       );
 
-      const accessoryEl = querySelector(this._element, '.Yh');
+      const accessoryEl = buttonOptions.buttonView.getElement(); // querySelector(this._element, '.Yh');
```

Comment:
yeah it's a bit weird that it works. but then why would `querySelector(this._element, '.Yh')` be what works? I assumed I was missing something somewhere

Reply:
ack or maybe it doesn't reliably fix it... I got it to work once

Reply:
the issue happens because the stream that updates descriptor is emitting two times and this code removes the previous icon before adding new icon

```
if (this._accessoryViewController) {
      this._accessoryViewController.destroy();
      this._accessoryViewController = null;
    }
    ```

Reply:
right, I thought it might be something like that but I'm not really sure what the best way to fix that kind of thing is

Reply:
>the issue happens because the stream that updates descriptor is emitting two times

it's emitting twice because mailfoo emits twice `pipeline loaded, button ready` kefir stream of passing nav item descriptor, the second time the stream reads oh we already have this `CREATE` type of accessory so let's remove it. the reason mail merge nav doesn't have the issue is because it's not using stream to pass in nav item descriptor. Now we need to find out:
1. has it always been like this? (emits twice on mailfoo)
2. verify emitting twice of the same stream is expecting to remove the same one

Reply:
It's the InboxSDK's fault if it doesn't handle multiple emits correctly. It should be fine for the application to pass an options stream that emits twice and there are many good reasons that Streak may do so.